### PR TITLE
Bug 1719315 - Add the URL metric type to the Glean schema

### PR DESCRIPTION
**Pull request by brizental**
Checklist for reviewer:

- [x] Commits should reference a bug or github issue, if relevant (if a bug is referenced, the pull request should include the bug number in the title)
- [ ] If adding a new field, the field should have a description (see #576 for an example)
- [ ] Scan the PR and verify that no changes (particularly to `.circleci/config.yml`) will cause environment variables (particularly credentials) to be exposed in test logs
- [ ] If the PR comes from a fork, trigger the `integration` CI test by pushing this revision [as discussed in the README](https://github.com/mozilla-services/mozilla-pipeline-schemas#packaging-and-integration-tests-optional) and review the report posted in the comments.

For glean changes:
- [ ] Update `templates/include/glean/CHANGELOG.md`

### CODE REVIEW

**Comment by jklukas** [comment]
Can you add a url to one of the validation documents? Perhaps https://github.com/mozilla-services/mozilla-pipeline-schemas/blob/main/validation/glean/glean.1.metrics.pass.json

**Comment by brizental** [comment]
> Can you add a url to one of the validation documents? Perhaps https://github.com/mozilla-services/mozilla-pipeline-schemas/blob/main/validation/glean/glean.1.metrics.pass.json

Hm, is it different from https://github.com/mozilla-services/mozilla-pipeline-schemas/pull/689/files#diff-80f957f65ca703bc9fd812ccf359f601bde1b9f21502bec19d44b370b646271bR90-R92?

**Comment by jklukas** [comment]
> Hm, is it different from https://github.com/mozilla-services/mozilla-pipeline-schemas/pull/689/files#diff-80f957f65ca703bc9fd812ccf359f601bde1b9f21502bec19d44b370b646271bR90-R92?

No, that is exactly what I wanted. This was a case of split attention leading to review of questionable quality...

File: templates/include/glean/url.1.schema.json [hunk 1]
```diff
@@ -0,0 +1,3 @@
+{
+  "type": "string"
```
Comment (jklukas) [inline templates/include/glean/url.1.schema.json:2]: Should we validate against a regex `pattern` here to enforce the same rules that are enforced at the SDK level per the [URL validation section of the proposal doc](https://docs.google.com/document/d/1kE4wV5qq-N4h5YZSft98-f3iVwt8RKkVWua8LblufC8/edit#heading=h.8tzmxuxhqe6m)?

Comment (brizental) [reply templates/include/glean/url.1.schema.json:2]: Hm, that sounds reasonable. I am actually changing validation logic a bit on the SDK side, once that is finished I'll add a new regex here.

File: templates/include/glean/glean.1.schema.json [hunk 2]
```diff
@@ -151,6 +151,10 @@
           @GLEAN_BASE_OBJECT_1_JSON@,
           "additionalProperties": @GLEAN_UUID_1_JSON@
         },
+        "url": {
+          @GLEAN_BASE_OBJECT_1_JSON@,
+          "additionalProperties": @GLEAN_STRING_1_JSON@
```
Comment (jklukas) [inline templates/include/glean/glean.1.schema.json:156]:
```suggestion
          "additionalProperties": @GLEAN_URL_1_JSON@
```

**Review by jklukas — COMMENTED** [review]
Regex looks good. Just need to make sure the new GLEAN_URL_1_JSON include is being used.

**Review by jklukas — APPROVED** [review]

**Pull request merged**